### PR TITLE
DCOS-21798 & DCOS-21614:  Backport to 1.11 

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
[Backport to 1.11 of this PR](https://github.com/dcos/dcos-ui/pull/2871)

## How to test

1. Add a new service with this configuration:

```
{
  "id": "/do-not-delete-me",
  "version": "2018-03-22T11:28:46.456Z",
  "containers": [
    {
      "name": "container-1",
      "resources": {
        "cpus": 0.1,
        "mem": 128,
        "disk": 0
      },
      "image": {
        "kind": "DOCKER",
        "id": "nginx"
      },
      "endpoints": [
        {
          "name": "http",
          "containerPort": 80,
          "hostPort": 0,
          "protocol": [
            "tcp"
          ]
        }
      ],
      "volumeMounts": [
        {
          "name": "home",
          "mountPath": "/etc/log"
        }
      ]
    }
  ],
  "volumes": [
    {
      "name": "home",
      "persistent": {
        "type": "root",
        "size": 10,
        "constraints": []
      }
    }
  ],
  "networks": [
    {
      "name": "dcos",
      "mode": "container"
    }
  ],
  "scaling": {
    "instances": 1,
    "kind": "fixed"
  },
  "scheduling": {
    "placement": {
      "constraints": []
    }
  },
  "executorResources": {
    "cpus": 0.1,
    "mem": 32,
    "disk": 10
  },
  "fetch": []
}
```

2. Visit the edit page of this service
3. Reload the edit page of this service and see there is no error

## Context
We accessed getLabel on the service without checking if the service is
even there. Therefore reloading the edit page caused an error.

By checking for a service and rendering an empty modal otherwise
we make sure that we don't try to render before we have the data.

Rendering the create form in these cases lead to a mutation of the data,
which in return lead to the edit form never being rendered.
Rendering null if there is no service lead to a dark flickering for the
time before the service loads. An empty modal gives a nicer UX.

**Checklist**
- [X] Did you add a JIRA issue in a commit message or as part of the branch name?
- [X] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [X] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

File: plugins/services/src/js/components/modals/__tests__/EditServiceModal-test.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { shallow } from "enzyme";
```
Comment (nLight) [inline plugins/services/src/js/components/modals/__tests__/EditServiceModal-test.js]: nope :trollface:

**Comment by DanielMSchmidt** [comment]
@nLight removed the test 🙈 if that's not ok for you I will try to rewrite it somehow

**Comment by jaxesn** [comment]
Maybe its just me, but should add a link to the original PR when doing these backport PRs?

**Comment by ahoskins** [comment]
run integration tests

**Comment by nLight** [comment]
@DanielMSchmidt can't we just add enzyme in 1.11? Old webpack is blocking that or something?

**Comment by DanielMSchmidt** [comment]
@nLight [I created this PR to backport enzyme only](https://github.com/dcos/dcos-ui/pull/2879) Once this is merged I will re-add the test here, therefore a hold merge for now

**Comment by DanielMSchmidt** [comment]
@nLight Update here, enzyme is now backported, I will redo this backport now 👍

**Comment by juliangieseke** [comment]
run integration tests

**Comment by DanielMSchmidt** [comment]
run integration tests

**Comment by DanielMSchmidt** [comment]
run integration tests

**Comment by nLight** [comment]
run integration tests

**Comment by nLight** [comment]
run integration tests

**Comment by ahoskins** [comment]
run integration tests

**Comment by nLight** [comment]
run integration tests

**Comment by DanielMSchmidt** [comment]
run integration tests

**Comment by nLight** [comment]
run integration tests

**Comment by DanielMSchmidt** [comment]
run integration tests

**Comment by DanielMSchmidt** [comment]
run integration tests

**Comment by juliangieseke** [comment]
its green :trollface:

**Pull request merged**